Services/SSLCertificateService.cs

```diff
@@ -196,11 +198,13 @@ public X509Certificate2 GetCertificate(StoreLocation storeLocation, StoreName st
                 {
                     if (IsCertificateMatchingNotNullParameters(certThumbPrint, friendlyName, certSubjectName, dnsName, notAfter, cert))
                     {
-                        return cert;
+                        result.Add(cert);
                     }
                 }
             }
-            return null;
+            if (!result.Any())
+                return null;
+            return result.OrderByDescending(cert=>cert.NotAfter).FirstOrDefault();
```

Comment:
This collection is an special collection, it is not accessible through LINQ as an enumerable